### PR TITLE
Added xpath function & Fix android launch issues

### PR DESCRIPTION
**Pull request by NathanaelA**
Fixed issues with android launches (You need to go directly to the actual build/apk path)  as it can pick up other files that you don't want...

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -94,3 +94,78 @@ exports.caps = {
...
+        case "slider": return "UISlider";
+        case "switch": return "UISwitch";
+        case "tabview": return "UITabBarItem";
+        case "textview": return "UITextView"
+        case "textfield": return "UITextField";
+        case "timepicker": return "UIDatePicker";
+        case "webview": return "UIWebView";
+    }
+
+    throw new Error("This "+name+" does not appear to tbe a standard NativeScript UI component");
```
Comment (joeskeen) [inline index.js]: small typo here: `appear to tbe a`

**Review by joeskeen — CHANGES_REQUESTED** [review]

**Comment by hdeshev** [comment]
The Android fix looks great. I don't understand what the XPath functions do though. Do we have to hook them up with Appium somehow?

**Comment by joeskeen** [comment]
@hdeshev, @NathanaelA explains this pretty well in his blog post: http://fluentreports.com/blog/?p=396

```js
// 'r' is the hearts pip
var heart = 'r';
        
// Setup our xpath
var xpath =  "//"+nsAppium.xpath("Label")+"[@text='"+heart+"']";

// Lets run our checks
return driver.elementByAccessibilityId("prior")
    .should.eventually.exist
    .tap().tap()
    .elementsByXPath(xpath).should.eventually.have.length(2);
```

> XPath actually allows you to drill down into the UI and find specific items that may have a specific hierarchy and/or certain parents.  For example; rather than search for all buttons, you can limit the search to buttons that are inside a GridLayout which is inside a StackLayout area.  However, the biggest downside with xpath is that it expects you to have the actual native android or native ios control type name.   For example; on Android the NativeScript Button class is actually using the android.widget.button.  The native class on iOS it is actually using an UIButton.   Now that makes XPath really, really hard to be cross platform test, doesn't it?     So to solve that issue; I have written a cool wrapper to help with this issue. It allows you to pass in your NativeScript class name and it will, depending on the platform you are testing against, will return the real underlying native component name.   So in this specific test case the xpath was "//" = any level of items, we aren't giving any specific parents (so find this anywhere in the layout).  Then my helper class nsAppium.xpath("Label") will give me the actual underlying UI name of a NativeScript Label component, and then finally "[@text='r']" means that element must have a "r" as the text field value.    In the case of the card it should find, the two pips on the edge of the card which should be a "r".  So this test would pass as long as the prior button worked to bring you to a King of Hearts card...

File: index.js (hunk 1 above)
Comment (hdeshev) [inline index.js]: Also, please fix formatting (spaces around operators, etc).

**Review by hdeshev — CHANGES_REQUESTED** [review]
The `xpath` functions seem to do nothing related to XPath. Can we rename them to reflect the actual usage. Some candidates off the top of my head:
- nativeView
- nativeViewName
- getNativeView
- nativeViewFor
- nativeElementName

**Comment by NathanaelA** [comment]
Actually their entire purpose is for the xpath function, example:

```
    var nsAppium = require("nativescript-dev-appium");
    var spade = 'q';
    var xpath =  "//"+nsAppium.xpath("Label")+"[@text='"+spade+"']";

    return driver.elementByAccessibilityId("card")
        .should.eventually.exist
        .tap()
        .elementsByXPath(xpath).should.eventually.have.length(3);
```

If you still dislike the name, pick one that you will accept and I'll make the change it to it.  I'm not hung up on the name...    

A couple other possible names:
.getXPathElement
.xPageFromNativeScript

File: index.js (hunk 1 above)
Comment (NathanaelA) [reply index.js:170]: I assume you are wanting then to be like this, where the returns all line up?:

```
case "x":     // here
case "yyyy":  // blah
case "nn":    // blah
```

**Comment by hdeshev** [comment]
OK, `getXPathElement` sounds fine -- let's go with it :)

I can see this feature getting better in the future if we can make it work for XPath queries using our own element names, perhaps using something like the [xpath-parser](https://github.com/dodo/xpath-parser) library.

**Comment by hdeshev** [comment]
We hit the apk pathing issue today, and had to merge your fix. You may need to rebase the PR to the latest master.

**Comment by NathanaelA** [comment]
This is superseded by #11 - Closing.

**Pull request closed without merging**